### DISCUSSION TITLE
Source code of lammps nvnmd plugin

### DISCUSSION THREAD

**Discussion by link89** [deepmodeling/deepmd-kit#2947] [General]
It looks like lammps plugin of nvnmd is not part of deepmd-kit. Is it in other repo or it is not open sourced? Is it possible for the community to customize it to support features like type_map, which is supported by the deepmd plugin?

Ref: 
* https://github.com/deepmodeling/deepmd-kit/blob/master/doc/nvnmd/nvnmd.md

**Comment by LiuGroupHNU** [comment]
We apologize, but NVNMD's pair_style is not open sourced. NVNMD can only be used with an FPGA device. As a result, only an expert with knowledge of FPGA hardware can update this code. You can ask us for additional features if you need them.

**Reply by link89** [reply]
Agree that FPGA is too hardcore to common users. But most of those change are naive tasks like mapping lammps atom type to dp's interal atom type, which happens in CPU before feeding data to FPGA accelerator.  Users should be able to help themselves for such cases.
